programs/manifest/tests/cases/global.rs

```diff
@@ -550,10 +555,151 @@ async fn global_insufficient() -> anyhow::Result<()> {
         test_fixture
             .global_fixture
             .global
-            .get_balance_atoms(&test_fixture.payer())
-            .unwrap(),
+            .get_balance_atoms(&test_fixture.payer()),
         0
     );
 
     Ok(())
 }
+
+#[tokio::test]
+async fn global_get_balance_not_in_global() -> anyhow::Result<()> {
+    let mut test_fixture: TestFixture = TestFixture::new().await;
+    let payer: Pubkey = test_fixture.payer();
+
+    test_fixture.global_fixture.reload().await;
+    let global_dynamic_account: DynamicAccount<GlobalFixed, Vec<u8>> =
+        test_fixture.global_fixture.global;
+
+    let balance_atoms: GlobalAtoms = global_dynamic_account.get_balance_atoms(&payer);
+    assert_eq!(balance_atoms, GlobalAtoms::ZERO);
+    Ok(())
+}
+
+#[tokio::test]
+async fn global_run_out_of_seats() -> anyhow::Result<()> {
+    let test_fixture: TestFixture = TestFixture::new().await;
+    let payer: Pubkey = test_fixture.payer();
+    let payer_keypair: Keypair = test_fixture.payer_keypair().insecure_clone();
+
+    for _ in 0..MAX_GLOBAL_SEATS {
+        let keypair: Keypair = Keypair::new();
+        // Fund gas for the account.
+        let _ = send_tx_with_retry(
+            Rc::clone(&test_fixture.context),
+            &[
+                transfer(&payer, &keypair.pubkey(), 10_000_000),
+                global_add_trader_instruction(&test_fixture.global_fixture.key, &keypair.pubkey()),
+            ],
+            Some(&payer),
+            &[&payer_keypair, &keypair],
+        )
+        .await;
+    }
+
+    // Last one doesnt work.
+    assert!(test_fixture.global_add_trader().await.is_err());
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn global_evict() -> anyhow::Result<()> {
+    let mut test_fixture: TestFixture = TestFixture::new().await;
+
+    let payer: Pubkey = test_fixture.payer();
+    let payer_keypair: Keypair = test_fixture.payer_keypair().insecure_clone();
+
+    for _ in 0..MAX_GLOBAL_SEATS - 1 {
```

Comment:
can we redefine this constant for the test to 5?

Reply:
Is there a good way to do that?

Reply:
needs to change in the code since we verify the global is full and block evict before then.